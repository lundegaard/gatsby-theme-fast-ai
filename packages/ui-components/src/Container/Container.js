import React from 'react';
import PropTypes from 'prop-types';

import useTheme from '../hooks/useTheme';
import Box from '../Box';

const Container = ({ fullWidth, ...rest }) => {
	const { grid: { gutters = 2 } = {}, breakpoints } = useTheme();

	return (
		<Box
			px={gutters}
			mx="auto"
			width={1}
			sx={{ maxWidth: fullWidth || !breakpoints ? 'none' : ['none', ...breakpoints] }}
			{...rest}
		/>
	);
};

Container.propTypes = { fullWidth: PropTypes.bool };

export default Container;
