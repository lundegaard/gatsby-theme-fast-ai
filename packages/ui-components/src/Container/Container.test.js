import React from 'react';
import { shallow } from 'enzyme';

import Container from './Container';

describe('<Container />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Container {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
