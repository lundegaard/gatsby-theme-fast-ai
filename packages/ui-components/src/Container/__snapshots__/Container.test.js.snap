// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> matches snapshot when shallowly rendered 1`] = `
<Box
  mx="auto"
  px={2}
  sx={
    Object {
      "maxWidth": Array [
        "none",
        undefined,
      ],
    }
  }
  width={1}
/>
`;
